src/App.tsx

```diff
@@ -1,16 +1,22 @@
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
+import { Movie } from './types/Movie';
 import moviesFromServer from './api/movies.json';
 
-export const App = () => {
+export const App: React.FC = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const addMovie = (movie: Movie) => setMovies([...movies, movie]);
```

Comment:
If you need to get state use callback
```suggestion
  const handleAddMovie = (movie: Movie) => setMovies(prevMovies => [...prevMovies, movie]);
```